### PR TITLE
docs: Add docs for onIncompatibleSchemaChange

### PR DESCRIPTION
**Pull request by danthelion**
**Description:**

(Describe the high level scope of new or changed features)

**Workflow steps:**

(How does one use this feature, and how has it changed)

**Documentation links affected:**

(list any [documentation links](https://docs.google.com/document/d/1SRC9VS9zyCzWl3n4HXHbc4wPB1eLxJHkA2rtu9ZNokM/edit?usp=sharing) that you created, or existing ones that you've identified as needing updates, along with a brief description)

**Notes for reviewers:**

(anything that might help someone review this PR)

- - -
This change is [Reviewable](https://reviewable.io/reviews/estuary/flow/1551)

### CODE REVIEW

**Review by psFried — COMMENTED** [review]
I've been slow to review this because the implementation is only half done. Today, auto-discovers don't actually respect or even look at `onIncompatibleSchemaChange`. I've been planning on addressing that as part of #1585, and just haven't gotten to it yet. These docs are accurate now, except that "when an automated action detects an incompatible schema change" is currently actually "never". So I'll propose holding off on merging this until #1585 merges, so we don't confuse people.

**Review by jgraettinger — APPROVED** [review]
LGTM

**Pull request merged**
